hs/src/Reach/Debugger.hs

```diff
@@ -0,0 +1,100 @@
+{-# OPTIONS_GHC -Wno-missing-export-lists #-}
+
+module Reach.Debugger where
+
+import Control.Monad.Reader
+
+import qualified Data.Map.Strict as M
+import Reach.AST.Base
+import Reach.AST.DLBase
+import Reach.AST.LL
+
+-- NOTE: let several base types be Integer/String as a placeholder for now
+
+type ConsensusEnv = M.Map DLVar DLVal
+
+type Balances = [Double]
+
+type Ledger = M.Map Account Balances
+
+type Frontend = ( Account, SLPart, M.Map DLVar DLVal, [ DLTail ] )
+
+type Frontends = [ Frontend ]
+
+type NewPartActions = M.Map SLPart [ DLTail ]
+
+type State = (Ledger, ConsensusEnv, Frontends, NewPartActions, Action)
+
+-- possible backend actions
+data Action
+  = ConsAction LLConsensus
+  | StepAction LLStep
+
+data Contract = Contract
+  { contr_contract_info :: Integer
+  , address :: String
+  , contr_token :: String
```

Comment:
What's the idea behind these? I don't think a contract will be explicitly represented, because the entire DApp is just one contract. In other words, the ConsensusEnv and the code pointer ARE the contract